Automattic/wp-calypso
# composite-checkout: Remove paymentData hook and callback argument

#### Changes proposed in this Pull Request

This is another attempt at #39095, except that this version is stand-alone and does not require another PR. 

This PR removes passing `paymentData` as an argument to `isCompleteCallback` and `isEditableCallback`. Those arguments were mostly unused and complicated both the implementation and the documentation of those functions.

This also removes the `usePaymentData` hook which was introduced as a simple way to store arbitrary data in payment methods and step objects. However, in practice it is far too inflexible to use for real data. Removing it should help to make it clear that the data layer of your custom components doesn't matter as far as this API is concerned (although `useRegisterStore` remains and option).

#### Testing instructions

- Run calypso with `ENABLE_FEATURES=composite-checkout-wpcom npm start`.
- Add a product to your cart and visit checkout.
- Make sure all steps display correctly.
- Make sure "Continue" button on each step is disabled when the step is incomplete.
- Make sure that the stripe card fields display some form of validation error when they are incorrect or are edited and then left empty, including the cardholder name.
- Make sure the postal code field displays a validation error if edited and then left empty.
- Make sure submit payment button is disabled until the final step.
- Make sure submit payment button is enabled on final step.
